### PR TITLE
Tab/Shift-Tab support for Widget_Vector

### LINKED ISSUE #1208: Widget_Vector should reacts to TAB key press
When editing value of vector parameters, eg layer `Origin`, user types `x` value first, press `Enter` (to activate/confirm edition) and then focus moves to `Y` value text field.

On the other hand, `TAB` (and `shift`+`TAB`)  is a common navigation key among form fields and this editor widget ignores it.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/89773521-widget_vector-should-reacts-to-tab-key-press?utm_campaign=plugin&utm_content=tracker%2F321991&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F321991&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by veermetri05**
Added support for Widget_Vector to respond to Tab and Shift-Tab as described in https://github.com/synfig/synfig/issues/1208

Close #1208

### CODE REVIEW

File: synfig-studio/src/gui/widgets/widget_vector.cpp [hunk 1]
```diff
@@ -315,6 +316,29 @@ Widget_Vector::show_all_vfunc()
...
+  if((key_event->keyval == GDK_KEY_ISO_Left_Tab)) {
+	  if(entry_y->is_focus()){
+		  entry_x->grab_focus();
+	  }
+	  else {
+		  Widget_Vector::activate();
+	  }
+  }
+  return Gtk::Box::on_key_press_event(key_event);
+}
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_vector.cpp:339]: You are mixing indentation here.
Please use `Tab` only for alignment in this case.

File: synfig-studio/src/gui/widgets/widget_vector.cpp [hunk 2]
```diff
@@ -315,6 +316,29 @@ Widget_Vector::show_all_vfunc()
...
+{
+  if(key_event->keyval == GDK_KEY_Tab)
+  { 
+    if(entry_x->is_focus()) { 
+		entry_y->grab_focus() ;
+	} 
+	else { 
+		Widget_Vector::activate() ;
+	}
+  }
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_vector.cpp:329]: Add `return true` right before finishing the `if` (same for the `if` below).

We don't want/need to call `Gtk::Box::on_key_press_event(key_event);` if we already handled the key event.

File: synfig-studio/src/gui/widgets/widget_vector.cpp [hunk 3]
```diff
@@ -315,6 +316,29 @@ Widget_Vector::show_all_vfunc()
...
+  if((key_event->keyval == GDK_KEY_ISO_Left_Tab)) {
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_vector.cpp:330]: As it was not so obvious for us, what about add an inline comment after `{`: `// Shift+Tab`?

**Review by rodolforg — CHANGES_REQUESTED** [review]
Thank you and congrats @veermetri05 :)
Please  address my comments here

**Comment by veermetri05** [comment]
I have done the requested changes, is there anything else ?

**Comment by ice0** [comment]
@veermetri05 Looks good, awaiting @rodolforg review.

**Comment by rodolforg** [comment]
@ice0 it's okay for me too.

Do you think he should change to `switch` instead of `if`?

Besides, we can squash and merge :)

**Comment by veermetri05** [comment]
@rodolforg Should I change to `switch` ? 

I don't think it would be good to use `switch` case, for now we are just checking `keyval`, **maybe** we need to use `state` sometimes then `switch` wouldn't be a good choice.

**Comment by rodolforg** [comment]
> **maybe** we need to use `state` sometimes then `switch` wouldn't be a good choice.

You could check `state` inside (or not) the `switch`. But it isn't essential/critical for now.

**Review by rodolforg — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**